tests/unit/checkmatelib/client_test.py

```diff
@@ -70,8 +70,37 @@ def test_it_with_a_bad_json_payload(self, client, response):
         with pytest.raises(CheckmateException):
             client.check_url("http://bad.example.com")
 
+    # pylint: disable=too-many-arguments
+    @pytest.mark.parametrize("prefix", ("http://", ""))
+    @pytest.mark.parametrize("path", ("/path", ""))
+    @pytest.mark.parametrize(
+        "domain,allowed",
+        (
+            # Local things we allow
+            ("localhost", True),
+            ("127.0.0.1", True),
+            ("0177.0000.0000.0001", True),  # 127.0.0.1 in octal
+            ("0.0.0.0", True),
+            # Local things we don't
+            ("192.168.0.1", False),
+            # Not public domains
+            ("my_private_name", False),
+            ("index.php", False),
```

Comment:
Might be worth noting briefly the scenario that motivated this `index.php` test case.